### PR TITLE
[FEATURE] Replace p_thread implementation with STL's thread support library

### PR DESCRIPTION
**Pull request by rrahn**

### CODE REVIEW

File: include/seqan/parallel/parallel_lock.h [hunk 1]
```diff
@@ -88,13 +88,14 @@ waitFor(SpinDelay & me)
...
-        sched_yield();
-#endif
+        std::this_thread::yield();
+//#ifdef STDLIB_VS
+//#if _WIN32_WINNT >= 0x0400
+//        SwitchToThread();
+//#endif
+//#else
+//        sched_yield();
+//#endif
```
Comment (rrahn) [inline include/seqan/parallel/parallel_lock.h]: Validate this.

File: include/seqan/parallel/parallel_queue_suspendable.h [hunk 2]
```diff
@@ -94,18 +91,17 @@ class ConcurrentQueue<TValue, Suspendable<TSpec> >
     TSize           back;
     TSize           front;
 
-    CriticalSection cs;
-    Condition       more;
+    std::mutex      cs;
+    std::condition_variable more;
```
Comment (rrahn) [inline include/seqan/parallel/parallel_queue_suspendable.h]: align data members.

File: include/seqan/parallel/parallel_queue_suspendable.h [hunk 3]
```diff
@@ -144,19 +140,17 @@ class ConcurrentQueue<TValue, Suspendable<Limit> >:
...
+        TBase()
     {
         reserve(this->data, maxSize, Exact());
         _setLength(this->data, maxSize);
     }
 
     ConcurrentQueue(ConcurrentQueue const & other):
-        TBase((TBase const &)other),
-        less(this->mutex)
+        TBase((TBase const &)other)
```
Comment (h-2) [inline include/seqan/parallel/parallel_queue_suspendable.h]: why this strange cast?

File: include/seqan/stream/iostream_bgzf.h [hunk 4]
```diff
@@ -706,9 +693,8 @@ class basic_unbgzf_streambuf :
                     // wait for the end of decompression
                     DecompressionJob &job = jobs[currentJobId];
                     {
```
Comment (h-2) [inline include/seqan/stream/iostream_bgzf.h]: introduce an empty line for readability

File: include/seqan/system/file_async.h [hunk 5]
```diff
@@ -1256,4 +1256,5 @@ deallocate(File<Async<TSpec> > const & me,
 
 }
 
-#endif
+#endif  // SEQAN_HEADER_FILE_ASYNC_H
+
```
Comment (h-2) [inline include/seqan/system/file_async.h]: why these changes in this file?

File: include/seqan/system/system_event.h [hunk 6]
```diff
@@ -37,6 +37,7 @@
 
 namespace seqan {
 
+// NOTE(rrahn): Only needed for File<Async> on Windows platforms.
```
Comment (h-2) [inline include/seqan/system/system_event.h]: document more clearly

File: include/seqan/system/system_thread.h [hunk 7]
```diff
@@ -220,6 +220,13 @@ namespace seqan
         return m.wait();
     }
 
+// TODO(rrahn): Thread creation with worker -> must be a callable.
+// TODO(rrahn): thread::join is wait() -> waitFor()
+// TODO(rrahn): no close or open? -> created/deleted on construction/destrucion
+// TODO(rrahn): We need detach
+// TODO(rrahn): Cancel aka close aka kill?
+
```
Comment (h-2) [inline include/seqan/system/system_thread.h]: ?

File: include/seqan/parallel/parallel_queue_suspendable.h (hunk 3 above)
Comment (rrahn) [inline include/seqan/parallel/parallel_queue_suspendable.h]: Did not touch this line. 
Only removed the ```less``` initialisation.

File: include/seqan/system/file_async.h (hunk 5 above)
Comment (rrahn) [inline include/seqan/system/file_async.h]: For better readability.

File: include/seqan/parallel/parallel_queue_suspendable.h [hunk 8]
```diff
@@ -189,32 +186,36 @@ template <typename TValue, typename TSpec>
...
 {
-    ScopedLock<CriticalSection> lock(me.cs);
+    std::unique_lock<std::mutex> lk(me.cs);
     if (--me.writerCount == 0u)
-        signal(me.more);
+    {
+        lk.unlock();
+        me.more.notify_all();
+    }
+
```
Comment (h-2) [inline include/seqan/parallel/parallel_queue_suspendable.h]: remove empty line

File: include/seqan/system/file_async.h (hunk 5 above)
Comment (h-2) [inline include/seqan/system/file_async.h]: still there is a newline too many at the EOF!

File: include/seqan/system/system_event_win.h [hunk 9]
```diff
@@ -348,6 +230,7 @@ inline bool signal(Event & e)
     return e.signal();
 }
 
+#endif  // #ifdef STDLIB_VS
 /*
```
Comment (h-2) [inline include/seqan/system/system_event_win.h]: can you remove the commented section below this since you are cleaning up already?

**Comment by h-2** [comment]
@rrahn should I merge this already?

**Comment by rrahn** [comment]
I'll remove the empty lines and merge by myself.

**Comment by rrahn** [comment]
Ok removed the comments and the empty lines.
I'll merge then.
